plugins/modules/ndo_service_device_cluster.py

```diff
@@ -0,0 +1,641 @@
+            - This parameter or O(interface_properties.interface.schema_id) is required.
+            type: str
+          schema_id:
+            description:
+            - ID of the schema associated with the interface.
+            - This parameter or O(interface_properties.interface.schema) is required.
+            type: str
+      interface_uuid:
+        description:
+        - UUID of the interface.
+        type: str
+      ipsla_monitoring_policy:
+        description:
+        - IPSLA monitoring policy configuration.
+        type: dict
+        suboptions:
+          name:
+            description:
+            - Name of the IPSLA monitoring policy.
+            type: str
+          template:
+            description:
+            - Template for the IPSLA monitoring policy.
+            - This parameter or O(interface_properties.ipsla_monitoring_policy.template_id) is required.
+            type: str
+          template_id:
+            description:
+            - ID of the template for the IPSLA monitoring policy.
+            - This parameter or O(interface_properties.ipsla_monitoring_policy.template) is required.
+            type: str
+      ipsla_monitoring_policy_uuid:
+        description:
+        - UUID of the IP SLA monitoring policy.
+        type: str
+      qos_policy:
+        description:
+        - Quality of Service (QoS) policy configuration.
+        type: dict
+        suboptions:
+          name:
+            description:
+            - Name of the QoS policy.
+            type: str
+          template:
+            description:
+            - Template for the QoS policy.
+            - This parameter or O(interface_properties.qos_policy.template_id) is required.
+            type: str
+          template_id:
+            description:
+            - ID of the template for the QoS policy.
+            - This parameter or O(interface_properties.qos_policy) is required.
+            type: str
+      qos_policy_uuid:
+        description:
+        - UUID of the QoS policy.
+        type: str
+      preferred_group:
+        description:
+        - Whether the interface belongs to a preferred group.
+        type: bool
+      rewrite_source_mac:
+        description:
+        - Whether to rewrite the source MAC address.
+        type: bool
+      anycast:
+        description:
+        - Indicates if anycast is enabled.
+        type: bool
+      config_static_mac:
+        description:
+        - Indicates if static MAC configuration is enabled.
+        type: bool
+      is_backup_redirect_ip:
+        description:
+        - Indicates if it is a backup redirect IP.
+        type: bool
+      load_balance_hashing:
+        description:
+        - Load balancing hashing method.
+        type: str
+        choices: ["sourceDestinationAndProtocol", "sourceIP", "destinationIP"]
+      pod_aware_redirection:
+        description:
+        - Indicates if pod-aware redirection is enabled.
```

Comment:
Should we add a bit more explanation on some of the features that can be enabled? Perhaps overkill so not required, just suggestion.